.github/workflows/ros2.yml

```diff
@@ -0,0 +1,101 @@
+name: build-and-bundle
+on: 
+  pull_request:
+  push:
+    branches:
+      - ros2
+  schedule: # (?) this will currently never trigger, since this workflow yaml file is not the default branch
+    - cron: '0 14 * * *'  # run at 2 pm everyday
+
+jobs:
+  build_and_bundle_ros2:
+    strategy:
+      matrix:
+        distro: ['dashing']
+        gazebo: [9]
+        include:
+        - distro: dashing
+          gazebo: 9
+          ubuntu_distro: bionic
+    runs-on: ubuntu-latest
+    name: 'Build and Bundle (ROS2)'
+    container:
+      image: rostooling/setup-ros-docker:ubuntu-${{ matrix.ubuntu_distro }}-ros-${{ matrix.distro }}-ros-base-latest
+    outputs:
+      robot_ws_build_result: ${{ steps.robot_ws_build.outcome }}
+      simulation_ws_build_result: ${{ steps.simulation_ws_build.outcome }}
+      # upload_result: ${{ steps.upload.outcome }}
+    steps:
+    - name: Checkout Branch
+      uses: actions/checkout@v1
+    - name: Setup Permissions
+      run: |
+        # TODO(setup-ros-docker#7): calling chown is necessary for now
+        sudo chown -R rosbuild:rosbuild "$HOME" .
+    - id: robot_ws_build
+      name: Build and Bundle Robot Workspace
+      uses: aws-robotics/aws-robomaker-github-actions/robomaker-sample-app-ci@2.0.0
+      with:
+        ros-distro: ${{ matrix.distro }}
+        gazebo-version: ${{ matrix.gazebo }}
+        workspace-dir: robot_ws
+        generate-sources: true
+    - id: simulation_ws_build
+      name: Build and Bundle Simulation Workspace
+      uses: aws-robotics/aws-robomaker-github-actions/robomaker-sample-app-ci@2.0.0
+      with:
+        ros-distro: ${{ matrix.distro }}
+        gazebo-version: ${{ matrix.gazebo }}
+        workspace-dir: simulation_ws
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID_ROS2 }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY_ROS2 }}
+        aws-region: ${{ secrets.AWS_REGION }}
+      if: always() && github.ref == 'refs/heads/ros2' # always for "push" and "schedule" (?) build status
+    # - id: upload
+    #   name: Upload to S3
+    #   uses: aws-robotics/aws-robomaker-github-actions/s3-cp-action@2.0.0
+    #   env:
+    #     AWS_S3_BUCKET: ${{ secrets.AWS_S3_BUCKET_ROS2 }}
+    #     AWS_REGION: ${{ secrets.AWS_REGION }}
+    #     FILES: 'sources.zip sources.tar.gz robot_ws.tar simulation_ws.tar'
+    #     DEST: 'travis/cloudwatch/${{ steps.robot_ws_build.outputs.ros-distro }}/${{ steps.robot_ws_build.outputs.gazebo-version }}/${{ steps.robot_ws_build.outputs.sample-app-version }}.${{ github.run_number }}/'
+    #   if: github.event_name == 'schedule' # upload to S3 on "schedule" (?) build
+
+  log_workflow_status_to_cloudwatch:
+    runs-on: ubuntu-latest
+    container:
+      image: ubuntu:bionic
+    needs:
+    - build_and_bundle_ros2
+    if: always()  # run even if the dependent jobs have failed to log failures
```

Comment:
Is this the convention you've been or will be following for the Cloud Extensions, to never log pull request results to CloudWatch Metrics?

Reply:
> nit-pick: you can actually omit the expression syntax per Github Actions documention

@ojasjoshi what did you mean here? Did you mean to use
```
if: always() && github.event_name != 'pull_request'
```
instead of
```
if: ${{ always() && github.event_name != 'pull_request' }}
```

Reply:
yep @mm318

Reply:
I think `always()` has side-effects; it is not a pure function. I'm concerned whether or not the expression syntax is there to negate some side-effects. Any thoughts, @emersonknapp?